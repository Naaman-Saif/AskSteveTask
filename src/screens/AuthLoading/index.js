import React from "react";
import { AsyncStorage, View } from "react-native";
import { Bubbles } from "react-native-loader";
import styles from "./styles";

export default class AuthLoadingScreen extends React.Component {
  constructor(props) {
    super(props);
    this._bootstrapAsync();
  }

  // Fetch the token from storage then navigate to our appropriate place
  _bootstrapAsync = async () => {
    const userToken = await AsyncStorage.getItem("User");

    // This will switch to the App screen or Auth screen and this loading
    // screen will be unmounted and thrown away.
    this.props.navigation.navigate(userToken ? "App" : "Auth");
  };

  // Render any loading content that you like here
  render() {
    return (
      <View style={styles.container}>
        <Bubbles size={10} color="#fec93b" />
      </View>
    );
  }
}
