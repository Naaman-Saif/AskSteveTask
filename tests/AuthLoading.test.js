jest.useFakeTimers();
import React from "react";
import AuthLoading from "../src/screens/AuthLoading";
import renderer from "react-test-renderer";

test("renders correctly", () => {
  const tree = renderer.create(<AuthLoading />).toJSON();
  expect(tree).toMatchSnapshot();
});
