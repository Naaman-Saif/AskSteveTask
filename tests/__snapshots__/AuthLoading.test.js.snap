// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ARTSurfaceView
    style={
      Array [
        undefined,
        Object {
          "height": 20,
          "width": 72,
        },
      ]
    }
  >
    <ARTShape
      d={
        Array [
          0,
          0,
          -10,
          4,
          0,
          0,
          10,
          -1.5707963267948966,
          1.5707963267948966,
          1,
          4,
          0,
          0,
          10,
          1.5707963267948966,
          -1.5707963267948966,
          1,
          1,
        ]
      }
      fill={
        Array [
          0,
          0.996078431372549,
          0.788235294117647,
          0.23137254901960785,
          1,
        ]
      }
      opacity={1}
      stroke={null}
      strokeCap={1}
      strokeDash={null}
      strokeJoin={1}
      strokeWidth={1}
      transform={
        Array [
          0,
          0,
          0,
          0,
          10,
          10,
        ]
      }
    />
    <ARTShape
      d={
        Array [
          0,
          0,
          -10,
          4,
          0,
          0,
          10,
          -1.5707963267948966,
          1.5707963267948966,
          1,
          4,
          0,
          0,
          10,
          1.5707963267948966,
          -1.5707963267948966,
          1,
          1,
        ]
      }
      fill={
        Array [
          0,
          0.996078431372549,
          0.788235294117647,
          0.23137254901960785,
          1,
        ]
      }
      opacity={1}
      stroke={null}
      strokeCap={1}
      strokeDash={null}
      strokeJoin={1}
      strokeWidth={1}
      transform={
        Array [
          0,
          0,
          0,
          0,
          36,
          10,
        ]
      }
    />
    <ARTShape
      d={
        Array [
          0,
          0,
          -10,
          4,
          0,
          0,
          10,
          -1.5707963267948966,
          1.5707963267948966,
          1,
          4,
          0,
          0,
          10,
          1.5707963267948966,
          -1.5707963267948966,
          1,
          1,
        ]
      }
      fill={
        Array [
          0,
          0.996078431372549,
          0.788235294117647,
          0.23137254901960785,
          1,
        ]
      }
      opacity={1}
      stroke={null}
      strokeCap={1}
      strokeDash={null}
      strokeJoin={1}
      strokeWidth={1}
      transform={
        Array [
          0,
          0,
          0,
          0,
          62,
          10,
        ]
      }
    />
  </ARTSurfaceView>
</View>
`;
